### ISSUE TITLE
ManyToMany multiple Relation Serious BUG: DELETE row in wrong table

### ISSUE THREAD

**Issue by marcellovetro** [strapi/strapi#1885]
**Informations**
- **Node.js version**: 10.9.0
- **npm version**: 6.4.1
- **Strapi version**: 3.0.0-alpha.14
- **Database**: MariaDB 10 
- **Operating system**: Ubuntu 16.04

**Issue**
In a scenario where you have more than one ManyToMany relation with the same table usin a Relational database such as MySql, the ContentManager make a wrong SQL that may delete the date from the wrong table. The problem is than the engine get wrongly the name of ManyToMany relation SQL table.

**Steps to reproduce the problem**
Create Content Types: Company, People, Takeover
- A Company have more people as Investors (ManyToMany Relation)
- A Takeover have more people as Investors (ManyToMany Relation)

If I delete an Investor in Company it should use the table company_investors to delete the record, instead it delete from the table operation_investors !!! 

The problem is exactly that both Entities have the same ManyToMany Relation field "Investors", if you rename the relation field it works right.

Wrong SQL, delete ALL occurrences!! : 
`delete from `takeover_investors` where `pmi_id` = ?'`

**What is the expected behavior?**

Expected SQL:
`delete from `company_investors` where `pmi_id` = ?' and `company_id` = ?'`

**Suggested solutions**

Check the function that return the relation table name ad Fix it

**Json Schema to reproduce the Issue**
Table: **Company**
```
{
  "connection": "default",
  "collectionName": "companies",
  "info": {
    "name": "company",
    "description": ""
  },
  "options": {
    "increments": true,
    "timestamps": true,
    "comment": ""
  },
  "attributes": {
    "Name": {
      "default": "",
      "type": "string"
    },
    "investors": {
      "collection": "people",
      "via": "companies",
      "dominant": true
    }
  }
}
```

Table: **People**
```
{
  "connection": "default",
  "collectionName": "people",
  "info": {
    "name": "people",
    "description": ""
  },
  "options": {
    "increments": true,
    "timestamps": true,
    "comment": ""
  },
  "attributes": {
    "Name": {
      "default": "",
      "type": "string"
    },
    "Email": {
      "default": "",
      "type": "email"
    },
    "takeOvers": {
      "collection": "takeover",
      "via": "investors"
    },
    "companies": {
      "collection": "company",
      "via": "investors"
    }
  }
}
```

Table: **Takeover**
```
{
  "connection": "default",
  "collectionName": "takeovers",
  "info": {
    "name": "takeover",
    "description": ""
  },
  "options": {
    "increments": true,
    "timestamps": true,
    "comment": ""
  },
  "attributes": {
    "Date": {
      "default": "",
      "type": "date"
    },
    "investors": {
      "collection": "people",
      "via": "takeOvers",
      "dominant": true
    },
    "Description": {
      "default": "",
      "type": "string"
    }
  }
}
```

**Comment by marcellovetro** [comment]
@lauriejim I want to fix that error but I need help to find the function that return the relation table name and the portion of code designed to synchronize the relations, could you or everyone read this issue tell me these references? Thanks!

**Comment by lauriejim** [comment]
@derrickmehaffy are you able to have this issue (I know you have many models and relations) ?

**Comment by derrickmehaffy** [comment]
@lauriejim I don't have any manyToMany models (I honestly couldn't find a usecase for it in my project :eyes: )

**Comment by marcellovetro** [comment]
I've updated the issue inserting the json schema

**Comment by lauriejim** [comment]
I'm starting investigation, it's really weird ^^

**Comment by lauriejim** [comment]
Normally this function is called:

https://github.com/strapi/strapi/blob/master/packages/strapi-hook-bookshelf/lib/relations.js#L71

then we enter in this case: 

https://github.com/strapi/strapi/blob/master/packages/strapi-hook-bookshelf/lib/relations.js#L142

**Comment by marcellovetro** [comment]
The problem is that at startup it loads only the first 20 items and if you do a search by writing in the input, it does not start the ajax to look for the key in the whole table. Adding my code has this behavior

**Comment by marcellovetro** [comment]
@lauriejim I wrote in the wrong issue :) I'm deleting this and the previuos message, sorry :)

**Comment by lauriejim** [comment]
@marcellovetro can you test the PR I did and confirm it works please.

**Comment by marcellovetro** [comment]
@lauriejim I've started yesterday to check the PR but I've found another problem that seem linked to that one, in two days I will check it and explain (or solve) the problem :)

**Comment by marcellovetro** [comment]
@lauriejim I've checked that your PR solve the problem, I've encounterd another problem if you have a manytomay relation with the same content type as a gerarchical table but I will create a dedicated issue describing the problem with example.

**Comment by marcellovetro** [comment]
Fixed by #1976
